src/variable_rate.jl

```diff
@@ -402,7 +429,7 @@ function (cache::VR_DirectEventCache)(u, t, integrator)
     return cache.current_threshold
 end
 
-@generated function execute_affect!(cache::VR_DirectEventCache{T, RNG, F1, F2}, integrator, idx) where {T, RNG, F1, F2 <: Tuple}
+@generated function execute_affect!(cache::VR_DirectEventCache{T, RNG, F1, F2}, integrator, idx) where {T, RNG, F1, F2}
```

Comment:
```suggestion
@generated function execute_affect!(cache::VR_DirectEventCache{T, RNG, F1, F2}, integrator, idx) where {T, RNG, F1, F2}
```
No, the previous version was correct as it is only for tuples and shouldn't be used for function wrappers. For function wrappers just evaluate the `idx` entry in the affect function array directly (and add a new dispatch here for that case).